lib/samples/cut_geometry/cut_geometry_sample.dart

```diff
@@ -0,0 +1,259 @@
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  // A flag to track if the geometry has been cut.
+  var _geometryCut = false;
+
+  // The graphics for the original state of the sample.
+  late final Graphic _lakeGraphic;
+  late final Graphic _borderGraphic;
+
+  // Graphics overlay to present the graphics for the sample.
+  final _graphicsOverlay = GraphicsOverlay();
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  crossAxisAlignment: CrossAxisAlignment.center,
+                  children: [
+                    // Button to trigger the cut function or reset to the original state.
+                    ElevatedButton(
+                      onPressed: _geometryCut ? resetGeometry : cutGeometry,
+                      child: _geometryCut
+                          ? const Text('Reset')
+                          : const Text('Cut'),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            Visibility(
+              visible: !_ready,
+              child: SizedBox.expand(
+                child: Container(
+                  color: Colors.white30,
+                  child: const Center(child: CircularProgressIndicator()),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() {
+    // Create a map with a topographic basemap style and add it to the map view controller.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = map;
+
+    // Add the graphics overlay to the map view controller.
+    _mapViewController.graphicsOverlays.add(_graphicsOverlay);
+
+    // Initialize the graphics for the sample.
+    initializeGraphics();
+
+    // Set the viewpoint to the lake graphic.
+    _mapViewController.setViewpoint(
+      Viewpoint.fromTargetExtent(_lakeGraphic.geometry!.extent),
+    );
+
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  // Initializes the graphics for the sample and adds them to the map.
+  void initializeGraphics() {
```

Comment:
I would fold this method into the body of onMapViewReady, so that it reads cleaner as a top-down tutorial. Though in contrast, the _initLakeGraphic and _initBorderGraphic make sense as separate methods because of those big blocks of data.